BuildScript.Nuke/BaseBuild.Documentation.cs

```diff
@@ -0,0 +1,220 @@
+  {
+    var documentationProjects = ReleaseProjectFiles.Where(projectMetadata => projectMetadata.IsDocumentationFile).ToList();
+    if (documentationProjects.Count == 0)
+      Log.Warning("No documentation projects found!");
+    documentationProjects.ForEach(projectFile =>
+    {
+      PrepareDocumentationProjectFile(projectFile);
+      try
+      {
+        GenerateSinglePackageWithDebugSymbols(projectFile);
+      }
+      catch (Exception ex)
+      {
+        Log.Error(ex, "Error while generating nuget package for documentation");
+      }
+      finally
+      {
+        RestoreDocumentationProjectFile(projectFile);
+      }
+    });
+  }
+
+  private void RestoreDocumentationProjectFile (ProjectMetadata projectFile)
+  {
+    FileSystemTasks.CopyFile($"{projectFile.ProjectPath}.bakup", projectFile.ProjectPath, FileExistsPolicy.Overwrite);
+    FileSystemTasks.DeleteFile($"{projectFile.ProjectPath}.bakup");
+  }
+
+  private void PrepareDocumentationProjectFile (ProjectMetadata projectFile)
+  {
+    FileSystemTasks.CopyFile(projectFile.ProjectPath, $"{projectFile.ProjectPath}.bakup", FileExistsPolicy.Overwrite);
+    var xmlDocument = new XmlDocument();
+    xmlDocument.Load(projectFile.ProjectPath);
+    var projectNode = xmlDocument.SelectSingleNode("/Project");
+    var itemGroupNode = xmlDocument.CreateElement("ItemGroup");
+    var noneNode = xmlDocument.CreateElement("None");
+    noneNode.SetAttribute("Pack", "True");
+    noneNode.SetAttribute("Include", DocumentOutputFile);
+    var copyToOutputNode = xmlDocument.CreateElement("CopyToOutputDirectory");
+    copyToOutputNode.InnerText = "PreserveNewest";
+    var packagePathNode = xmlDocument.CreateElement("PackagePath");
+    packagePathNode.InnerText = $"doc/{AssemblyMetadata.DocumentationFileName}.chm";
+    noneNode.AppendChild(packagePathNode);
+    noneNode.AppendChild(copyToOutputNode);
+    itemGroupNode.AppendChild(noneNode);
+    projectNode!.AppendChild(itemGroupNode);
+    xmlDocument.Save(projectFile.ProjectPath);
+  }
+  private IReadOnlyCollection<ProjectMetadata> PrepareProjectMetadata (IReadOnlyCollection<ProjectMetadata> projectMetadataList)
+  {
+    return projectMetadataList.SelectMany(
+        projectMetadata =>
+            projectMetadata.TargetFrameworks.Select(
+                targetFramework => 
+                    CreateProjectMetadataByTargetFramework(projectMetadata, targetFramework))
+    ).ToList();
+  }
+
+  private ProjectMetadata CreateProjectMetadataByTargetFramework (ProjectMetadata projectMetadata, string targetFramework)
+  {
+    if (!projectMetadata.IsMultiTargetFramework)
+    {
+      return projectMetadata;
+    }
+    return new ProjectMetadata
+           {
+               Configuration = projectMetadata.Configuration,
+               ProjectPath = projectMetadata.ProjectPath,
+               ToolsVersion = projectMetadata.ToolsVersion,
+               IsMultiTargetFramework = projectMetadata.IsMultiTargetFramework,
```

Comment:
When stripping the additional framework metadata, this should be set to `false` imo, just to make sure the semantics meet the expectations.